File: code-review-graph/tests/fixtures/sample.pl
use strict;
use warnings;
use File::Basename;

package Animal;

sub new {
    my ($class, %args) = @_;
    return bless \%args, $class;
}

sub speak {
    my ($self) = @_;
    return "...";
}

package Dog;

sub new {
    my ($class, %args) = @_;
    my $self = Animal::new($class, %args);
    return $self;
}

sub fetch {
    my ($self, $item) = @_;
    return "Fetched $item";
}

sub bark {
    my ($self) = @_;
    print $self->speak() . "\n";
}

File: code-review-graph/tests/fixtures/sample.dart
import 'dart:async';
import 'package:flutter/material.dart';

abstract class Animal {
  String get name;
  void speak();
}

mixin SwimmingMixin {
  void swim() => print('swimming');
}

enum PetType { dog, cat, bird }

class Dog extends Animal with SwimmingMixin {
  final String name;
  final PetType type;

  Dog(this.name) : type = PetType.dog;

  @override
  void speak() {
    print('Woof! I am $name');
  }

  Future<void> fetch(String item) async {
    await _run();
    print('Fetched $item');
  }

  void _run() {
    print('running');
  }

  static Dog create(String name) {
    return Dog(name);
  }
}

Dog createDog(String name) {
  return Dog(name);
}
